test/sqllogictest/transform/aggregation_nullability.slt

```diff
@@ -158,25 +168,35 @@ EXPLAIN WITH(arity, join implementations) select t1.f1, count(t2.f1), sum(t2.f1)
 ----
 Explained Query:
   Return // { arity: 9 }
-    Filter (#6 >= 0) // { arity: 9 }
-      Reduce group_by=[#0] aggregates=[count(#2), sum(#2), max(#2), min(#2), count(#1), sum(#1), min(#1), max(#1)] // { arity: 9 }
-        Union // { arity: 3 }
-          Map (null) // { arity: 3 }
-            Union // { arity: 2 }
-              Negate // { arity: 2 }
-                Project (#0, #1) // { arity: 2 }
-                  Join on=(#0 = #2) type=differential // { arity: 3 }
-                    implementation
-                      %1[#0]UKA » %0:l0[#0]K
-                    Get l0 // { arity: 2 }
-                    ArrangeBy keys=[[#0]] // { arity: 1 }
-                      Distinct project=[#0] // { arity: 1 }
-                        Project (#0) // { arity: 1 }
-                          Get l1 // { arity: 2 }
-              ReadStorage materialize.public.t1 // { arity: 2 }
-          Project (#0, #1, #0) // { arity: 3 }
-            Get l1 // { arity: 2 }
+    Project (#0..=#2, #6, #7, #3, #4, #8, #9) // { arity: 9 }
+      Filter (#4 >= 0) // { arity: 10 }
```

Comment:
It looks like this should be able to be pushed down to the first `Reduce`, essentially becoming `HAVING SUM(#1) >= 0`.

Reply:
It's not pushed in MIR because that would prevent arrangement re-use, so `JoinImplementation` actually lifts it away, I guess.

(But I guess it's pushed during the mir-to-lir lowering into the closure that is _after_ the corresponding join stage.)

Reply:
Ok, it feels like there's maybe a tension that we want the arrangement output from `Reduce`, but also we know how to push filters into `Reduce`; does the LIR translation do that level of MFP pushdown (through the entire `Join` operator, and on to the next operator). It did not when I originally wrote it, iirc, and if that's still the case perhaps we want to think about pushing it down despite the join arrangement interactions.

Reply:
No, the filter will currently end up inside the LIR Join node.

I guess we could add a special case in `JoinImplementation`: If an already-arranged input is a Reduce, then there is no need to lift away the Filter, because arrangement reuse will happen even with the Filter being left there, because lowering will push the Filter into the `mfp_after` of the `Reduce`. Do I understand this correctly?

Reply:
I think the only case we don't want to push down the filter is when it is a `Get` operator with an arrangement it references? If it is a bespoke `ArrangeBy` .. we want to push down the filter. If it is a `Reduce` we should push it down. If it is a `Threshold` or a `TopK`, .. I don't think we make those arrangements available .. but if we did we could push down a filter to them.